src/Widgets/WindowOverview.vala

```diff
@@ -24,19 +22,12 @@ public class Gala.WindowOverview : Clutter.Actor, ActivatableComponent {
     }
 
     construct {
-        display = wm.get_display ();
-        display.get_workspace_manager ().workspace_switched.connect (() => { close (); });
-        display.restacked.connect (restack_windows);
+        wm.get_display ().get_workspace_manager ().workspace_switched.connect (() => { close (); });
```

Comment:
From what I remember, connecting to static objects never get disconnected so one has to manually do that (but that has to be verified)